tests/System.Text.Primitives.Tests/EncodingPerfTestsCoreCLR.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text.Utf8;
+using Xunit;
+using Microsoft.Xunit.Performance;
+
+namespace System.Text.Primitives.Tests
+{
+    public partial class EncodingPerfComparisonTests
+    {
+        [Benchmark]
+        [InlineData(DataLength, 0x0, Utf8OneByteLastCodePoint)]
+        [InlineData(DataLength, Utf8OneByteLastCodePoint + 1, Utf8TwoBytesLastCodePoint)]
+        [InlineData(DataLength, Utf8TwoBytesLastCodePoint + 1, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, Utf16HighSurrogateFirstCodePoint, Utf16LowSurrogateLastCodePoint)]
+        [InlineData(DataLength, 0x0, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.AlternatingASCIIAndNonASCII)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.MostlyASCIIAndSomeNonASCII)]
+        public void EncodeFromUtf8toUtf8CoreCLR(int length, int minCodePoint, int maxCodePoint, SpecialTestCases special = SpecialTestCases.None)
+        {
+            string inputString = GenerateStringData(length, minCodePoint, maxCodePoint, special);
+            char[] characters = inputString.AsSpan().ToArray();
+            Encoding utf8 = Encoding.UTF8;
+            int utf8Length = utf8.GetByteCount(characters);
+            var utf8Buffer = new byte[utf8Length];
+            utf8.GetBytes(characters, 0, characters.Length, utf8Buffer, 0);
+
+            var output = new byte[utf8Length];
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                    output = Encoding.Convert(utf8, utf8, utf8Buffer);
+            }
+        }
+
+        [Benchmark]
+        [InlineData(DataLength, 0x0, Utf8OneByteLastCodePoint)]
+        [InlineData(DataLength, Utf8OneByteLastCodePoint + 1, Utf8TwoBytesLastCodePoint)]
+        [InlineData(DataLength, Utf8TwoBytesLastCodePoint + 1, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, Utf16HighSurrogateFirstCodePoint, Utf16LowSurrogateLastCodePoint)]
+        [InlineData(DataLength, 0x0, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.AlternatingASCIIAndNonASCII)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.MostlyASCIIAndSomeNonASCII)]
+        public void EncodeFromUtf16toUtf8CoreCLR(int length, int minCodePoint, int maxCodePoint, SpecialTestCases special = SpecialTestCases.None)
+        {
+            string inputString = GenerateStringData(length, minCodePoint, maxCodePoint, special);
+            char[] characters = inputString.AsSpan().ToArray();
+            Encoding utf8 = Encoding.UTF8;
+            int utf8Length = utf8.GetByteCount(characters);
+            var utf8Buffer = new byte[utf8Length];
```

Comment:
The test is measuring the performance of that exact function call and I do that within the loop being measured. Using  Encoding.UTF8.GetBytes(inputString) causes an allocation of the byte[] being returned.

```C#
            int utf8Length = utf8.GetByteCount(characters);
            var utf8Buffer = new byte[utf8Length];

            foreach (var iteration in Benchmark.Iterations)
            {
                using (iteration.StartMeasurement())
                    utf8.GetBytes(characters, 0, characters.Length, utf8Buffer, 0);
            }
```